internal/services/containers/kubernetes_nodepool.go

```diff
@@ -218,7 +218,12 @@ func SchemaDefaultNodePool() *pluginsdk.Schema {
 						Optional: true,
 						ForceNew: true,
 					},
-
+					"host_group_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
```

Comment:
Validate func refs to https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/compute/linux_virtual_machine_resource.go#L156